Improve user experience & AI readability of the Azure Container Apps Instrumentation page by migrating to Customizable Docs (#38440)
DataDog/documentation: content/en/serverless/azure_container_apps/_index.mdoc.md

Reviewer: nit: if theres a way, itd be great to only have a `$` appear on the first line, or remove the `\` and see if we can have line wrapping for these commands:
<img width="678" height="195" alt="Image" src="https://github.com/user-attachments/assets/d0a7970c-62ae-41c7-b4dd-e945e43f976c" />

    text: 'Build secure and scalable Azure serverless applications with the Well-Architected Framework'
  - link: "/integrations/azure/"
    tag: "Documentation"
    text: "Azure Integration"
---

{% collapse-content title="In-Container vs. Sidecar" level="h5" %}

| Aspect | In-Container | Sidecar |
|--------|--------------|---------|
| Deployment | One container (your app, wrapped with the Datadog Agent) | Two containers (your app, Datadog Agent) |
| Image changes | Increases app image size. | No change to app image. |
| Cost overhead | Less than sidecar (no extra container). | Extra vCPU/memory. Overallocating the sidecar wastes cost; underallocating leads to premature scaling. |
| Logging | Direct stdout/stderr access. | Shared volume + log library routing to a log file. Uncaught errors require extra handling, since they are not automatically handled by your logging library. |
| Failure isolation | In rare cases, Datadog Agent bugs can affect your app. | Datadog Agent faults are isolated. |
| Best for | Simpler setup, lower cost, and direct log piping. | Multiple containers per service, Agent isolation, and performance-sensitive workloads. |

{% /collapse-content %}

> ## Setup

Reviewer: Should this page still state the Azure Integration prerequisite before setup? The further-reading card may be sufficient, but the previous in-container and sidecar landing pages explicitly made it the first step.

Author: Ah you are right. At least for this PR, I think it would be best to preserve that. I will make the change

Author: done!

  - /serverless/azure_container_apps/in_container/dotnet/
  - /serverless/azure_container_apps/in_container/ruby/
  - /serverless/azure_container_apps/in_container/php/
  - /serverless/azure_container_apps/sidecar/
  - /serverless/azure_container_apps/sidecar/python/
  - /serverless/azure_container_apps/sidecar/nodejs/
  - /serverless/azure_container_apps/sidecar/go/
  - /serverless/azure_container_apps/sidecar/java/
  - /serverless/azure_container_apps/sidecar/dotnet/
  - /serverless/azure_container_apps/sidecar/ruby/
  - /serverless/azure_container_apps/sidecar/php/
further_reading:
  - link: "/serverless/guide/disable_serverless"
    tag: "Documentation"
    text: "Disable Serverless Monitoring"
  - link: 'https://www.datadoghq.com/blog/azure-container-apps/'
    tag: 'Blog'
    text: 'Collect traces, logs, and custom metrics from Container Apps services'
  - link: 'http://datadoghq.com/blog/azure-well-architected-serverless-applications-best-practices/'
    tag: 'Blog'
    text: 'Build secure and scalable Azure serverless applications with the Well-Architected Framework'
  - link: "/integrations/azure/"
    tag: "Documentation"
    text: "Azure Integration"
---

> {% collapse-content title="In-Container vs. Sidecar" level="h5" %}

Reviewer 2: Nothing wrong with the existing approach but it would be more typical to just have an overview section. I do think we are missing a brief overview telling users about this page. Switching from a collapsed section avoids starting the page with an h5 as well, which could hurt the page structure for agents.
```
## Overview

Azure Container Apps is a fully managed serverless platform for deploying and scaling
containerized applications. Datadog monitors Container Apps in two layers:

- The Azure integration collects standard metrics and logs.
- The Datadog `serverless-init` Agent adds distributed tracing, enhanced metrics, custom
  metrics, and direct log collection.

Instrument your application with `serverless-init` in one of two ways:

- **In-container**: Wraps your application container with the Datadog Agent. Choose this
  for a simpler setup, lower cost overhead, and direct log piping.
- **Sidecar**: Runs the Datadog Agent in a separate container alongside your application
  container. Choose this if you have multiple containers in a single service, if you prefer
  strict isolation of the Datadog Agent, or if you have performance-sensitive workloads.
```

Author: made changes to add a short overview and repositioned the collapsed section.

| Aspect | In-Container | Sidecar |
|--------|--------------|---------|

> | Deployment | One container (your app, wrapped with the Datadog Agent) | Two containers (your app, Datadog Agent) |

Reviewer 2: Keeping punctuation consistent through the table.
```suggestion
| Deployment | One container (your app, wrapped with the Datadog Agent). | Two containers (your app, Datadog Agent). |
```

| Image changes | Increases app image size. | No change to app image. |

> | Cost overhead | Less than sidecar (no extra container). | Extra vCPU/memory. Overallocating the sidecar wastes cost; underallocating leads to premature scaling. |

Reviewer 2: ```suggestion
| Cost overhead | Less than sidecar (no extra container). | Extra vCPU/memory. Overallocating the sidecar wastes resources; underallocating leads to premature scaling. |
```

> | Logging | Direct stdout/stderr access. | Shared volume + log library routing to a log file. Uncaught errors require extra handling, since they are not automatically handled by your logging library. |

Reviewer 2: ```suggestion
| Logging | Direct stdout/stderr access. | Requires a shared volume and log library routing to a log file. Uncaught errors require extra handling, since they are not automatically handled by your logging library. |
```

| Failure isolation | In rare cases, Datadog Agent bugs can affect your app. | Datadog Agent faults are isolated. |
| Best for | Simpler setup, lower cost, and direct log piping. | Multiple containers per service, Agent isolation, and performance-sensitive workloads. |

{% /collapse-content %}

> ## Setup

Reviewer: One other thing an agent found:

The previous page documented that Serverless Agent instrumentation provides enhanced metrics, linked to the metrics reference, and identified the `azure.app_containerapps.enhanced.*` namespace. Could we restore that guidance here? Without it, users can no longer discover the enhanced metrics capability or know which namespace to query.

Author: done!

**Prerequisite**: Set up the [Azure Integration](/integrations/azure/) to collect metrics and logs from Azure Container Apps.

> The setup below instruments your application with the Datadog `serverless-init` Agent to enable tracing, enhanced metrics, custom metrics, and direct log collection. [Enhanced metrics](/integrations/azure-container-apps/#metrics) are distinguished with the `azure.app_containerapps.enhanced.*` namespace.

Reviewer 2: ```suggestion
The following instruments your application with the Datadog `serverless-init` Agent to enable tracing, enhanced metrics, custom metrics, and direct log collection. [Enhanced metrics](/integrations/azure-container-apps/#metrics) use the `azure.app_containerapps.enhanced.*` namespace.
```

```dockerfile {% filename="Dockerfile" %}
COPY --from=datadog/serverless-init:<YOUR_TAG> /datadog-init /app/datadog-init
ENTRYPOINT ["/app/datadog-init"]
CMD ["/nodejs/bin/node", "/path/to/your/app.js"]
```

{% collapse-content title="Alternative configuration" level="h4" %}
Datadog expects `serverless-init` to be the top-level application, with the rest of your app's command line passed in for `serverless-init` to execute.

If you already have an entrypoint defined inside your Dockerfile, you can instead modify the CMD argument.

```dockerfile
CMD ["/app/datadog-init", "/nodejs/bin/node", "/path/to/your/app.js"]
```

If you require your entrypoint to be instrumented as well, you can instead swap your entrypoint and CMD arguments.

```dockerfile
ENTRYPOINT ["/app/datadog-init"]
CMD ["/your_entrypoint.sh", "/nodejs/bin/node", "/path/to/your/app.js"]
```

As long as your command to run is passed as an argument to `datadog-init`, you receive full instrumentation.
{% /collapse-content %}
{% /step %}
{% step title="Set up logs" %}
To enable logging, set the environment variable `DD_LOGS_ENABLED=true`. This allows `serverless-init` to read logs from stdout and stderr.

> Datadog also recommends setting the environment variables `DD_LOGS_INJECTION=true` and `DD_SOURCE=nodejs` to enable advanced Datadog log parsing.

Reviewer 2: I believe DD_LOGS_INJECTION is for trace/log correlation whereas DD_SOURCE is for parsing? This is repeated a few times on the page. I think it would be more precise to say:
> Set DD_LOGS_INJECTION=true to correlate logs and traces, and DD_SOURCE=node.js to enable language-specific parsing.
